File: client-app/src/app/models/patient.ts
import { Allergy } from "./allergy";
import { CBC } from "./cbc";
import { Doctor } from "./doctor";
import { doctorprofile } from "./doctorprofile";
import { Height } from "./height";
import { LiverPanel } from "./liverpanel";
import { MetabolicPanel } from "./metabolicpanel";
import { Prescription } from "./prescription";
import { Symptoms } from "./symptoms";
import { Urinalysis } from "./urinalysis";
import { WaterIntake } from "./waterintake";
import { Weight } from "./weight";

export interface Patient {
    id:string;
    name: string;
    lastName: string;
    birthDate: string;
    address:string;
    language:string;
    profession:string;
    city :string;
    area:string;
    information :string;
    number :string;
    bloodGroup:string; 
    doctor:doctorprofile;
    prescriptions: Prescription[];
    allergies: Allergy[];
    cbCs: CBC[];
    liverPanels:LiverPanel[];
    metabolicPanels:MetabolicPanel[];
    urinalysisList: Urinalysis[];
    symptoms: Symptoms[];
    weight : Weight [];
    height : Height [];
    waterIntake : WaterIntake [];
    
}

export class Patient implements Patient {
    constructor(init?: PatientFormValues) {
      Object.assign(this, init);
    }
  }

  export class PatientFormValues {
    id?:string = undefined;
    name: string = '';
    lastName: string = '';
    birthDate: string= '';
    address:string = '';
    language:string = '';
    profession:string ='';
    city :string = '';
    area:string = '';
    information :string = '';
    number :string = '';
    bloodGroup:string ='';

    constructor(patient?: PatientFormValues) {
      if (patient) {
        this.id = patient.id;
        this.name = patient.name;
        this.lastName = patient.lastName;
        this.birthDate = patient.birthDate;
        this.city = patient.city;
        this.language = patient.language;
        this.area = patient.area;
        this.profession=patient.profession;
        this.number = patient.number;
        this.bloodGroup = patient.bloodGroup;
        this.address = patient.address;
        this.information = patient.information

      }
    }
  }

File: Domain/Patient.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class Patient : AppUser
    {
        public string Name { get; set; }

        public string LastName {get; set;}

        public DateTime? BirthDate {get; set;}

        public string Address{get; set;}

        public string Language {get; set;}

        public string Profession { get; set; }

        public string City { get; set; }

        public string Area { get; set; }
        
        public string Information {get; set;}

        public string Number {get;set;}

        public string BloodGroup {get; set;}

        public Doctor doctor {get; set;}

        public ICollection<Height> Height { get; set; } = new List<Height>();

        public ICollection<Weight> Weight { get; set; } = new List<Weight>();

        public ICollection<WaterIntake> WaterIntake { get; set; } = new List<WaterIntake>();

        public ICollection<Prescription> Prescriptions {get; set;} = new List<Prescription>();

        public ICollection<Allergy> Allergies {get; set;} = new List<Allergy>();

        public ICollection<Vitals> Vitals {get; set;} = new List<Vitals>();

         public ICollection<Appointment> Appointments {get; set;} = new List<Appointment>();

        public ICollection<CBC> CBCs {get; set;} = new List<CBC>();

        public ICollection<Urinalysis> UrinalysisList {get; set;} = new List<Urinalysis>();

        public ICollection<MetabolicPanel> MetabolicPanels {get; set;} = new List<MetabolicPanel>();

         public ICollection<LiverPanel> LiverPanels {get; set;} = new List<LiverPanel>();

         public ICollection<Symptoms> Symptoms {get; set;} = new List<Symptoms>();


    }
}